doc/ConfigurationHeader.md

```diff
@@ -0,0 +1,45 @@
+Configuration Header for *cupla*
+=============================
+
+cupla provides configuration header to use the library without CMake, e.g in a [cling](https://github.com/root-project/cling) environment.
+
+
+Available Pre-compiler Definitions 
+=================================
+
+Pre-compiler definitions can be used to control the behavior of cupla/alpaka.
+The definitions must be passed via a compiler flag or be defined before the accelerator header is included.
+
+- `CUPLA_STREAM_ASYNC_ENABLED`: `0` use synchronous streams, `1` use asynchronous streams
```

Comment:
What is the default behaviour is the symbol is not defined ?

Reply:
the default for `CUPLA_STREAM_ASYNC_ENABLED` is `0` https://github.com/ComputationalRadiationPhysics/cupla/blob/6a97c1cf1140ea76da0ec72d3803f5c63a66c7ad/include/cupla/types.hpp#L136-L140

I will extend the documentation string